projects/ad9361/src/ad9361_tinyiiod.c

```diff
@@ -0,0 +1,2721 @@
+			  const struct channel_info *channel)
+{
+	return (ssize_t) sprintf(buf, "%d", 0); /* dummy */
+}
+
+/**
+ * get_ensm_mode
+ * @param *buff where value is stored
+ * @param len maximum length of value to be stored in buf
+ * @param *channel channel properties
+ * @return length of chars written in buf, or negative value on failure
+ */
+ssize_t get_ensm_mode(char *buf, size_t len,
+		      const struct channel_info *channel)
+{
+	ssize_t ret = 0;
+	ret = ad9361_ensm_get_state(ad9361_phy);
+	if (ret < 0)
+		return ret;
+	if (ret >= ARRAY_SIZE(ad9361_ensm_states) ||
+	    ad9361_ensm_states[ret] == NULL) {
+		return -EIO;
+	}
+
+	return sprintf(buf, "%s", ad9361_ensm_states[ret]);
+}
+
+/**
+ * get_filter_fir_config
+ * @param *buff where value is stored
+ * @param len maximum length of value to be stored in buf
+ * @param *channel channel properties
+ * @return length of chars written in buf, or negative value on failure
+ */
+ssize_t get_filter_fir_config(char *buf, size_t len,
+			      const struct channel_info *channel)
+{
+	return sprintf(buf, "FIR Rx: %d,%d Tx: %d,%d",
+		       ad9361_phy->rx_fir_ntaps, ad9361_phy->rx_fir_dec,
+		       ad9361_phy->tx_fir_ntaps, ad9361_phy->tx_fir_int);
+}
+
+/**
+ * get_calib_mode
+ * @param *buff where value is stored
+ * @param len maximum length of value to be stored in buf
+ * @param *channel channel properties
+ * @return length of chars written in buf, or negative value on failure
+ */
+ssize_t get_calib_mode(char *buf, size_t len,
+		       const struct channel_info *channel)
+{
+	uint8_t en_dis;
+	ad9361_get_tx_auto_cal_en_dis(ad9361_phy, &en_dis);
+
+	return (ssize_t) snprintf(buf, len, "%s", en_dis ? "auto" : "manual");
+}
+
+/**
+ * read all attributes from map
+ * @param *buff where value is stored
+ * @param len maximum length of value to be stored in buf
+ * @param *channel channel properties
+ * @param *map map with coresponding attributes
+ * @param map_size
+ * @return length of chars written in buf
+ */
+static ssize_t read_all_attr(char *buf, size_t len,
+			     const struct channel_info *channel, const struct attrtibute_map* map,
+			     int32_t map_size)
+{
+	int16_t i, j = 0;
+	char local_buf[0x1000];
+	for(i = 0; i < map_size; i++) {
+
+		int16_t attr_length = map[i].exec((local_buf), len, channel);
+		int32_t *len = (int32_t *)(buf + j);
+		*len = Xil_EndianSwap32(attr_length);
+
+		j += 4;
+		if(attr_length >= 0) {
+			sprintf(buf + j, "%s", local_buf);
+			if (attr_length & 0x3) /* multiple of 4 */
+				attr_length = ((attr_length >> 2) + 1) << 2;
+			j += attr_length;
+		}
+	}
+
+	return j;
+}
+
+/**
+ * write all attributes from map
+ * @param *buff where value is stored
+ * @param len maximum length of value to be stored in buf
+ * @param *channel channel properties
+ * @param *map map with coresponding attributes
+ * @param map_size
+ * @return length of chars written in buf
+ */
+static ssize_t write_all_attr(char *buf, size_t len,
```

Comment:
Just to double-check: is this supposed to be empty?

Reply:
The corresponding function in iio-oscilloscope for "write_all_attr" is:
update_from_ini()->iio_channel_attr_write_all()
Since iio-oscilloscope doesn't support, for now, loading a profile from .osc_profile.ini, when serial interface is used, "write_all_attr" is not implemented yet. This should be implemented in the future.